### ISSUE TITLE
Using name:es to the whole Spain is wrong

### ISSUE THREAD

**Issue by ravilacoya** [organicmaps/organicmaps#8730]
Some regions of Spain have different official languages. For example, in Galicia it is the gl, in Catalunya is the ca, etc.

At the Spanish country level, those names are the only official ones, so even in official documents you won't see Lérida (name:es) instead of Lleida (name:ca, the official one), nor Orense (name:es) instead of Ourense (name:gl, the official one). While on the road, you will see only Lleida and Ourense, never Lérida or Orense.

Changing the language of the phone to Galician (gl) didn't fix this problem.

To fix this issue, I would strongly suggest that you use the OSM name=* tag for the whole Spain, or that the app changes to gl when the phone is in Galician, to ca when the phone is in Catalan and to eu when the phone is in Euskera. I suggested using Organic Maps to my sister in law, and she came to me quite surprised when she saw the names of places in Galicia in the old Spanish translation, that is really wrong and linked to the times of dictatorship moreover.

**Comment by biodranik** [comment]
@ravilacoya what is the system language on your sister-in-law's phone? Is it Spanish? If yes, that explains why she sees everything in Spanish. Did she try to set the Galician system language?

**Comment by dvdmrtnz** [comment]
I have been thinking about this and we could definitely improve here. Take a look at what we show when app is set to English vs Spanish for places in the Basque Country that are tagged with `name`, `name:es` and `name:en`:

| English | Spanish |
| ------------- | ------------- |
| Title = `name:en` | Title = `name:es` |
| Subtitle =`name` | Subtitle = empty |
| ![](https://github.com/user-attachments/assets/5e6636de-020c-4502-9bbb-03a8684c5565) | ![](https://github.com/user-attachments/assets/9a2c9c75-e4af-4e75-9d05-6e0036b536ed) |

I would expect to see the subtitle as the `name` too when the app is set to Spanish, because that is what is usually shown in signs in the real world.

https://www.openstreetmap.org/node/123561791#map=19/42.99706/-2.42785

**Comment by dvdmrtnz** [comment]
Another example here. I would expect to see the `name` tag also written below the `name:es` title when the app is in Spanish:

App set to English/Spanish:

<img src="https://github.com/user-attachments/assets/30f703c2-a0d1-4944-b7c7-c6aab416b8f7" width="320"/> <img src="https://github.com/user-attachments/assets/1c1d3bb2-3ba5-45a2-ad93-d8d4a3dfdcd0" width="320"/>

https://www.openstreetmap.org/node/4194898831#map=19/43.31729/-1.97765

**Comment by biodranik** [comment]
Can the issue be fixed by properly marking languages in data/countries_meta.txt (if they're supported in coding/string_utf8_multilang.cpp ?

**Comment by dvdmrtnz** [comment]
I don't think so. For example `Spain_Comunidad Foral de Navarra` is already correctly set to `es` and `eu` in `data/countries_meta.txt` and it doesn't show the basque language below the spanish one when the app is in Spanish. 

App set to English/Spanish:


<img src="https://github.com/user-attachments/assets/7b9d59d2-edec-46ef-8c17-7eb4b849d61c" width="320"/> <img src="https://github.com/user-attachments/assets/258c7b8f-c4d7-4457-91d2-4562254969e4" width="320"/>

https://www.openstreetmap.org/node/4036206773#map=17/42.96155/-1.67859

**Comment by biodranik** [comment]
When both es and eu are set, it means that the `name` tag values are mixed and may be in both eu and es languages. Is it correct? Shouldn't this region lang be set to eu only?

**Comment by Crashillo** [comment]
IMHO, the best option would be allowing the users to choose the display language in the app settings, or picking the local names instead, as well.

The assumption of _one country, one language_ is a terrible mistake, since usually leads to war editions. It's also important to remark that the rest of spain' languages are not so wide-spread as the spanish, tech-wise speaking. That implies, for instance, you could set your phone in spanish, in order to use another apps which lack of your regional language, but you do want to see the map in your regional language.

As @ravilacoya said, in Spain, the _official_ names are different from the _spanish_ names. One thing is the country, another thing is the language.

> Shouldn't this region lang be set to eu only?

Some regions have a [default_language](https://wiki.openstreetmap.org/wiki/Key:default%20language) key, that you could use as well, but in the case of Navarra this default language is partial (some parts are basque-speaking, some mixed, some spanish-speaking... and the region of Asturias partially speaks _asturian_, Aragon partially speaks _aragonese_, etc...). As you can see, in terms of programming, that's a mess.

That's why I suggest to let the users themselves to set their wished language, better than picking the mobile phone one. And as a mapper, the sooner we face this issue the lesser conflictive editions we have, as you see, is a sensitive problem.

**Comment by muralito** [comment]
> IMHO, the best option would be allowing the users to choose the display language in the app settings, or picking the local names instead, as well.

I agree. But choosing a "wise" default is also important for usability.

The diversity of languages in Spain is one reasons for this kind of solution, but probably not the only one.
Probably a spanish user want to set the preferences according to their needs or beliefs, but a foreign user traveling Spain, unaware of all this conflicts, just want to use the map which is best suited for their purpose. E.g. the UI or OS preference is in english, but does he/she wants the street name in english (assuming they are tagged name:en), or would want "name" to see what is in the ground?

E.g. Me traveling to countries in Asia or Middle East, where not only the languages are different and I'm not used to, but the alphabet is absolutely different and the symbols looks almost the same, and also in other places the reading is left to right. ¿What solution is the best for such case? (No, not going there is not a solution :-) )

**Comment by muralito** [comment]
> Just a note: Organic Maps is already available in Euskara and it shows names from `name:eu` when data is available if you select that language in you phone settings. Same for Català.
> <img alt="" width="320" src="https://private-user-images.githubusercontent.com/47610359/350404496-3f29cc79-479d-4633-87ad-f780816c665d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ2MTA1MTAsIm5iZiI6MTczNDYxMDIxMCwicGF0aCI6Ii80NzYxMDM1OS8zNTA0MDQ0OTYtM2YyOWNjNzktNDc5ZC00NjMzLTg3YWQtZjc4MDgxNmM2NjVkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjE5VDEyMTAxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQzM2MxOGIxZjhhOGU2YTFmYWU4N2U4YTc1N2UyY2U2ZWJlOGQwYTA1MDVmNTU0Zjg3NjJjZDNmYzkzZjAxM2MmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.4jWTlTLUmtZYIowHeRrLW4F_JxLuuWsugItfUvMMv70">

The source of the problem seems to be that the convention in Spain is to use the type of way in the name :-) but that is just like it is. E.g. from your screenshot, "Larrazko", that in name has "calle" or "kalea" (spanish and euskera words for "street"). Getting calle/kalea out of the name, gives the same name no matter if spanish or euskera. :-)
And seeing the street signs in that neighbourhood, with big font for name, and very small ones to "calle" and "kalea".

BTW, how do you usually search for a streets in OM, typing "Calle Lazarko","Lazarko Kalea" or just "Lazarko" and OM does the work?

**Comment by dvdmrtnz** [comment]
> Getting calle/kalea out of the name, gives the same name no matter if spanish or euskera. :-)

That’s not always the case. See for example: "Avenida Sancho el Fuerte" vs "Antso Azkarraren etorbidea":

<img src="https://github.com/user-attachments/assets/b850b74f-942a-4297-a1f9-1b693c528764" width="320"/> <img src="https://github.com/user-attachments/assets/274e29ee-52fa-4344-b79b-0b4006dcfb9e" width="320"/>

**Comment by muralito** [comment]
It's beautiful to see that OM can put all OSM data in use to show each user according to their preferences.
